sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/HttpClient.java

```diff
@@ -20,11 +22,38 @@ public interface HttpClient {
     Response<?> send(HttpRequest request);
 
     /**
-     * Creates a new {@link HttpClient} instance.
+     * Creates an instance of the default {@link HttpClient} implementation, entirely unrelated to all other instances
+     * that were created previously.
      *
-     * @return A new {@link HttpClient} instance.
+     * @return An instance of the default {@link HttpClient} implementation, entirely unrelated to all other instances
+     * that were created previously.
      */
-    static HttpClient createDefault() {
-        return HttpClientProviders.createInstance();
+    static HttpClient getDefault() {
+        return new DefaultHttpClientBuilder().build();
+    }
+
+    /**
+     * Returns a new instance of the {@link HttpClient} that the default {@link HttpClientProvider} is configured to
+     * create. If no {@link HttpClientProvider} can be found on the classpath, an instance of the default
+     * {@link HttpClient} implementation will be created instead, in which it would effectively be the same as calling
+     * {@link #getDefault()}.
+     *
+     * @return A new {@link HttpClient} instance, entirely unrelated to all other instances that were created
+     * previously.
+     */
+    static HttpClient getInstance() {
+        final String NO_DEFAULT_PROVIDER_MESSAGE = "A request was made to load the default HttpClient provider "
+            + "but one could not be found on the classpath. If you are using a dependency manager, consider including "
+            + "a dependency on azure-core-http-netty or azure-core-http-okhttp. Depending on your existing "
+            + "dependencies, you have the choice of Netty or OkHttp implementations. Additionally, refer to "
+            + "https://aka.ms/azsdk/java/docs/custom-httpclient to learn about writing your own implementation.";
```

Comment:
Need to fix the message here

Reply:
Yes, the message should not refer to Azure packages and probably should have a different aka.ms link as the guidance for implementing the client may be different for generic client.

Reply:
We can update the aka.ms links in a future PR after we've created the required documentation.

Reply:
For now I've removed all other mentions of Azure or things we don't currently support, like Netty.